src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticationBuilder.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    internal class RemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>
+        : IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>
+        where TRemoteAuthenticationState : RemoteAuthenticationState
+        where TAccount : RemoteUserAccount
+    {
+        public RemoteAuthenticationBuilder(IServiceCollection services) => Services = services;
+
+        public IServiceCollection Services { get; }
+    }
```

Comment:
Do we need to have the `IRemoteAuthenticationBuilder` interface? This seems like an unopinionated enough concrete type that we could make it `public` and not have the interface. Is there any reason why a different top-level implementation of the interface would ever exist?

Reply:
This is consistent with the way it is done in ASP.NET Core, although to be fair, there are some differences between different areas of ASP.NET Core too. I followed the approach that MVC and other areas I'm familiar with follow (and we had a discussion at the beginning of asp.net core to homogenize this, which resulted in me doing some changes across the stack).

That said, if you feel strongly about it, I'm happy to just use a base class for this.

Reply:
I don't feel strongly about it.